### PR TITLE
Implement Homepage 2.2

### LINKED ISSUE #522: Implement Homepage 2.2
Currently, the bounce rate for the Spotlight ND homepage is somewhere close to 85%, and is similar to this for the PA homepage. Ideally, every person who visits the homepage should at least explore one narrative. 

To fix this, we should implement a redesign of the page that decreases the bounce rate. The redesign should do the following: 
- Give users a better understanding of the data that is in the site. 
- Pull the calls-to-action (CTAs) above the fold of the page. 
- Achieve both of the above objectives on mobile 

#### Proposed Design

Figma for the below design [here](https://www.figma.com/file/kejdI2WbdTFJU). 

The component to the right is a carousel; every 5 seconds, the next item in the carousel should highlight. The items should be identical to what is currently on the Spotlight homepage, including: 
- Prison Population over Time
- Racial Disparities: Population by Race/Ethnicity
- Probation: Revocations from Probation, By Type
- Parole: Programming Participation

#### Measurement
**Bounce rate** should be measured before and after this change to observe a difference. A more thorough approach would involve A/B testing the two designs, but the current audience of Spotlight is likely too small to observe a significant difference over a reasonable timespan.

### PR DESCRIPTION
**Pull request by nasaownsky**
## Description of the change

Implementing Homepage 2.2

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

Closes #522 

### Development

These boxes should be checked by the submitter prior to merging:

- [ ] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: spotlight-client/src/OtherNarrativeLinksPreview/OtherNarrativeLinksPreview.tsx [hunk 1]
```diff
@@ -141,58 +135,67 @@ const RacialDisparitiesPreview = observer(
...
+    PREVIEW_ORDER.indexOf(DEFAULT_SELECTED_TAB)
+  );
+
+  React.useEffect(() => {
+    const nextIndex = (tabIndex + 1) % (PREVIEW_ORDER.length - 1);
+
+    const timer = setTimeout(() => {
+      setTabIndex(nextIndex);
+      selectTab(PREVIEW_ORDER[nextIndex]);
+    }, 5000);
```
Comment (nasaownsky) [inline spotlight-client/src/OtherNarrativeLinksPreview/OtherNarrativeLinksPreview.tsx:167]: I personally think, that 5 seconds is not enough for user to try interact with preview. Maybe double it? Try it for yourself @hobuobi @terryttsai

**Comment by nasaownsky** [comment]
Also, @terryttsai maybe you know where to disable start animation of `WindowedTimeSeries` chart? It seems a bit clunky, and I think that we better should be using standard opacity transition in the preview. Thanks!

**Review by hobuobi — CHANGES_REQUESTED** [review]
Awesome work! So exciting to see this implemented. A few comments here and there: 
### Major/Breaking
- The current implementation does not work with fewer than four visualizations (e.g., on Pennsylvania's homepage, it tries to rotate to an imaginary fourth visualization and shows nothing for five seconds). 
- I noticed a strange bug on the other pages where the legends for the bubble charts seemed to overlap with the actual visualization... not sure why, but not seeing the same thing in staging or production!

### Design Nits
- Use the typical body font for the text below the header (for ND, "At the department of corrections in North Dakota, we believe that sharing information builds greater accountability between us and the communities we serve.")
- Font size of the carousel labels ("Prison," "Racial Disparities," etc) could be larger. 1.5rem seemed like a possible good size on desktop. 
- Button: 
    - Font size of the button is too small; should be about the same size as the carousel labels (suggested 1.5rem). 
    - Should add a shadow to the button. (Design spec is #006C67 at 0.3 opacity; Y-offset of 10; blur of 20px. Can play around with it a little!) 
    - Ideal would be to have the button width flex based on its contents, rather than stay a static width. 
- The margins seem inconsistent (wider) with the Narrative pages. 
- Can we see what this would look like if we added the legends back? 

This should cover almost everything I caught! Thanks again for the hard work.

**Comment by terryttsai** [comment]
Trying this out, when I wave my cursor over the right-most racial disparity segment, the popover is cut-off. 
Screen Shot 2022-04-26 at 4 23 03 PM
Screen Shot 2022-04-26 at 4 23 44 PM

Also, is there supposed to be no popover in the parole breakdown by county?
Screen Shot 2022-04-26 at 4 22 53 PM

**Comment by hobuobi** [comment]
Awesome work! To wrap some of this up: 
### RESPONSES TO THE ABOVE
- re: chart overflow: @terryttsai already mentioned a possible solution to the bubble chart overflow issue. If that doesn't work or breaks other things, I think it's minor enough to skip. 
- re: margins: the left and right margins for the homepage are slightly smaller than on the narrative pages. But also minor enough to skip, so I wouldn't worry about it if it's not a completely trivial fix!

### DESIGN NITS
- Reduce the size of the header on mobile. 1.75rem seems to work well. 
- Body text on mobile homepage looks like the right size, but is _slightly_ too small on desktop. Would bump this up to be around 1.3rem on desktop. 
- Is it possible to ease/transition the width of the button as it changes size? Lower priority, so if it’s hard to make it robust, no worries. 
- Add "Data" after each section name to the button (e.g. "Explore Racial Disparities Data"). Please make sure the button still fits everywhere!
- Center the button on mobile.

### FUNCTIONALITY 
- **We should implement a “pause” for the carousel:** If the user interacting with a chart, the timer should pause and the carousel should not advance to the next section until the user stops interacting with the chart. “Interacting” means that they are hovering over the chart or the legend.
- Links should open in a new tab. 

### CONTENT
Please copy exactly as written. 
#### US_ND
- Header: Our mission is to transform lives, influence change, and strengthen community.
- Body: At the [North Dakota Department of Corrections and Rehabilitation](http://docr.nd.gov/), transparency is a critical element of our mission; sharing information builds greater accountability between the DOCR and the communities we serve.
#### US_PA
- Header: [The Pennsylvania Department of Corrections (DOC)](https://www.cor.pa.gov/) is committed to enhancing public safety.
- Body: The DOC's mission is to reduce criminal behavior by providing individualized treatment and education to incarcerated individuals, resulting in successful community reintegration through accountability and positive change.
#### US_ID (subject to change, but good placeholder)
- Header: Explore data from Idaho’s corrections system.
- Body: The [Idaho Department of Corrections](https://www.idoc.idaho.gov/) aims to protect the public, our staff, and those within our custody and supervision through safety, accountability, partnerships and providing opportunities for change.
#### US_TN (subject to change, but good placeholder)
- Header: Explore data from Tennessee’s corrections system.
- Body: The [Tennessee Department of Correction](https://www.tn.gov/correction.html) has an inherent responsibility to provide those within our custody with opportunities to grow. Sharing information builds greater accountability between our department and the communities we serve.

File: spotlight-client/src/charts/BubbleChart.tsx [hunk 2]
```diff
@@ -72,14 +72,20 @@ const LegendWrapper = styled.div`
...
 
 export default function BubbleChart({
   data,
   height,
+  showLegend,
 }: BubbleChartProps): React.ReactElement {
   const { highlighted, setHighlighted } = useHighlightedItem();
 
+  if (!showLegend) {
+    margin.bottom = 0;
```
[the remark this answers is not in the document]
Comment (terryttsai) [reply spotlight-client/src/charts/BubbleChart.tsx:86]: Generating the margins in this component would ensure this doesn't happen: `const margin = { top: 0, left: 0, right: 0, bottom: showLegend ? 40 : 0 };`

**Comment by nasaownsky** [comment]
I don't think dynamically changing a width of an element depending on its content is possible, as there is no transition for content. The content of an element changes instantly and the width does not get a numerical value in this case - but adjusts.

**Review by hobuobi — APPROVED** [review]
Thank you for addressing all of the comments! I think this is all good from a product/design POV; will leave it to @terryttsai to wrap up code review. 
CC @agronomic in case he wants to check out the new design/functionality. :)

**Review by terryttsai — APPROVED** [review]
LGTM!

**Pull request merged**